packages/react-dom-bindings/src/client/ReactDOMFloatClient.js

```diff
@@ -795,8 +969,104 @@ function createPreloadResource(
   };
 }
 
+function acquireHeadResource(resource: HeadResource): Instance {
+  resource.count++;
+  let instance = resource.instance;
+  if (!instance) {
+    const {props, root, type} = resource;
+    switch (type) {
+      case 'title': {
+        const titles = root.querySelectorAll('title');
+        for (let i = 0; i < titles.length; i++) {
+          if (titles[i].textContent === props.children) {
+            instance = resource.instance = titles[i];
+            markNodeAsResource(instance);
+            return instance;
+          }
+        }
+        instance = resource.instance = createResourceInstance(
+          type,
+          props,
+          root,
+        );
+        insertResourceInstanceBefore(root, instance, titles.item(0));
+        break;
+      }
+      case 'meta': {
+        let existingEl = null;
+        let insertBefore = null;
+
+        const metaResource: MetaResource = (resource: any);
+        const {matchAttr, matchValue, parentResource} = metaResource;
+
+        if (matchAttr === 'charset') {
+          existingEl = root.querySelector('meta[charset]');
+        } else {
+          let scope: Document | Element = root;
+          let parent = null;
+          let parentProperty = null;
+          if (matchAttr === 'property' && parentResource) {
+            const parentInstance = parentResource.instance;
+            if (parentInstance && parentInstance.parentElement) {
+              parent = parentInstance;
+              insertBefore = parent.nextSibling;
+              parentProperty = parentResource.matchValue;
+              scope = parentInstance.parentElement;
+            }
+          }
+          const metas = scope.querySelectorAll('meta');
```

Comment:
One possible idea is to generate a query similar to what you do for `acquireStyleResource` but generate it in `getResource` and you use that as the key.

So the key you use for the map *is* the selector. That way it's reusable for many resources. You could do `parentKey + ',' + childKey` to find both.

I'm a little concerned about this being n * n traversals for number of meta tags which can be quite a lot since this runs for every meta tag. Not sure a more specific query selector helps though.